pkg/gui/controllers/helpers/fixup_helper.go

```diff
@@ -77,15 +77,29 @@ func (self *FixupHelper) HandleFindBaseCommitForFixupPress() error {
 		return self.c.ErrorMsg("The base commit for this change is already on master")
 	}
 
-	if !useIndex {
-		if err := self.c.Git().WorkingTree.StageAll(); err != nil {
-			return err
+	doIt := func() error {
+		if !useIndex {
+			if err := self.c.Git().WorkingTree.StageAll(); err != nil {
+				return err
+			}
+			_ = self.c.Refresh(types.RefreshOptions{Mode: types.SYNC, Scope: []types.RefreshableView{types.FILES}})
 		}
-		_ = self.c.Refresh(types.RefreshOptions{Mode: types.SYNC, Scope: []types.RefreshableView{types.FILES}})
+
+		self.c.Contexts().LocalCommits.SetSelectedLineIdx(index)
+		return self.c.PushContext(self.c.Contexts().LocalCommits)
 	}
 
-	self.c.Contexts().LocalCommits.SetSelectedLineIdx(index)
-	return self.c.PushContext(self.c.Contexts().LocalCommits)
+	if hasHunksWithOnlyAddedLines {
+		return self.c.Confirm(types.ConfirmOpts{
+			Title:  self.c.Tr.FindBaseCommitForFixup,
+			Prompt: self.c.Tr.HunksWithOnlyAddedLinesWarning,
+			HandleConfirm: func() error {
+				return doIt()
+			},
+		})
+	}
+
+	return doIt()
```

Comment:
why do we need the doIt function? Seems like it would behave the same if inlined

Reply:
Na I'm fine with this pattern